### PR TITLE
Refactor `OasstApiClient`

### PR DESCRIPTION
**Pull request by notmd**
- Refactor to avoid duplicated code.
- Fixed a few typing issues.
- I moved the prettier config to the root of the website folder to allow formating cypress test code.

### CODE REVIEW

**Comment by fozziethebeat** [comment]
Can you check the build issue?

**Review by fozziethebeat — CHANGES_REQUESTED** [review]
Overall I like this refactoring, especially the types.  I'll approve once I see that the contract tests pass

**Review by fozziethebeat — APPROVED** [review]

File: website/.prettierignore
Comment (fozziethebeat) [inline website/.prettierignore:1]: This didn't seem to have an effect on pre-commit.  I can later exclude these from pre-commit checks.

**Comment by fozziethebeat** [comment]
Merge from main.  I just made a change to the pre-commit check to allow your change here.

File: website/.prettierignore
Comment (notmd) [reply website/.prettierignore:1]: Yeah, dont know why running prettier with pre-commit doesn't apply the ignore config.

**Comment by notmd** [comment]
> Merge from main. I just made a change to the pre-commit check to allow your change here.

I just updated the pre-commit config to respect the `.prettierignore` file.

**Comment by notmd** [comment]
The CI failed is not related to my change.

**Review by fozziethebeat — APPROVED** [review]
Due to the pre commit change you'll need Yannic's review before we can merge.

**Review by yk — APPROVED** [review]
lgtm, thank you

**Pull request merged**
